Kitodo/src/main/java/org/kitodo/production/helper/WikiFieldHelper.java

```diff
@@ -109,4 +118,172 @@ private static String addMatchingTagMessage(String type) {
+                        }
+                    }
+                } else if (message.contains("Orange K")) {
+                    comment.setType(CommentType.ERROR);
+                    comment.setCorrected(Boolean.TRUE);
+                    Property correctionPerformed = getCorrectionPerformedProperty(process, message);
+                    if (Objects.nonNull(correctionPerformed)) {
+                        comment.setCreationDate(correctionPerformed.getCreationDate());
+                        try {
+                            comment.setCorrectionDate(getCreationDate(correctionPerformed));
+                            deleteProperty(process, correctionPerformed);
+                        } catch (ParseException | DAOException | DataException e) {
+                            Helper.setErrorMessage("Deleting error", logger, e);
+                        }
+                    }
+                    comment.setCurrentTask(ServiceManager.getProcessService().getCurrentTask(process));
+                    comment.setCorrectionTask(getWikiFieldCorrectionTask(message, process));
+                } else {
+                    comment.setType(CommentType.INFO);
+                }
+                try {
+                    ServiceManager.getCommentService().saveToDatabase(comment);
+                    process.setWikiField("");
+                    ServiceManager.getProcessService().save(process);
+                } catch (DAOException | DataException e) {
+                    Helper.setErrorMessage("Saving error", logger, e);
+                }
+            }
+        }
+    }
+
+    private static Property getCorrectionRequiredProperty(Process process, String message) {
+        List<Property> properties = process.getProperties();
+        for (Property property : properties) {
+            String translation = Helper.getTranslation("correctionNecessary");
+            String msg = getWikiFieldMessage(message);
+            if (property.getTitle().equals(translation)
+                    && property.getValue().contains(msg)) {
+                return property;
+            }
+        }
+        return null;
+    }
+
+    private static Property getCorrectionPerformedProperty(Process process, String message) {
+        List<Property> properties = process.getProperties();
+        for (Property property : properties) {
+            if (property.getTitle().equals(Helper.getTranslation("correctionPerformed"))
+                    && property.getValue().endsWith(getWikiFieldCorrectionTask(message, process).getTitle())) {
+                return property;
+            }
+        }
+        return null;
+    }
+
+    private static Task getCorrectionTask(Property property) throws DAOException {
+        int correctionTaskId = Integer.parseInt(property.getValue().substring(property.getValue().indexOf(" CorrectionTask: ") + 17,
+                property.getValue().indexOf(")")));
```

Comment:
Use single Quotes ')' instead of double quotes ")" to use the faster indexOff(char) method